### PR TITLE
Add methods which allow client get schemas info

### PR DESCRIPTION
**Pull request by evidolob**
### What does this PR do?
It add LSP methods which allow Client get information about all know schemas and schemas used to validate current file.
That methods allows to build UI for schema selection.

### What issues does this PR fix or reference?
https://github.com/redhat-developer/vscode-yaml/issues/565

### Is it tested? How?
With tests

### CODE REVIEW

File: src/languageservice/utils/schemaUrls.ts [hunk 1]
```diff
@@ -22,3 +24,48 @@ export function checkSchemaURI(
...
+ * Collect all urls of sub schemas
+ * @param schema the root schema
+ * @returns map url to schema
+ */
+export function getSchemaUrls(schema: JSONSchema): Map<string, JSONSchema> {
+  const result = new Map<string, JSONSchema>();
+  if (!schema) {
+    return result;
+  }
+  const url = schema.url;
```
Comment (gorkem) [inline src/languageservice/utils/schemaUrls.ts:38]: Init the constant inside the following `if` statement or not create a constant at all.

File: src/languageservice/utils/schemaUrls.ts [hunk 2]
```diff
@@ -22,3 +24,48 @@ export function checkSchemaURI(
...
+  if (url) {
+    if (url.startsWith('schemaservice://combinedSchema/')) {
+      addSchemasForOf(schema, result);
+    } else {
+      result.set(schema.url, schema);
```
Comment (gorkem) [inline src/languageservice/utils/schemaUrls.ts:43]: Use `url` instead of `schema.url`

File: README.md [hunk 3]
```diff
@@ -254,6 +254,90 @@ docker run -it quay.io/redhat-developer/yaml-language-server:latest
 
 `yaml-language-server` use `vscode-languageserver@7.0.0` which implements [LSP 3.16](https://github.com/Microsoft/language-server-protocol/blob/gh-pages/_specifications/specification-3-16.md)
 
+## Language Server Protocol extensions
+
+### SchemaSelectionRequests
+
+#### SupportSchemaSelection Notification
+
+The support schema selection notification is sent from a client to the server to inform server that client support JSON Schema selection.
```
Comment (gorkem) [inline README.md:263]: ... that the client support -> support**s**

File: README.md [hunk 4]
```diff
@@ -254,6 +254,90 @@ docker run -it quay.io/redhat-developer/yaml-language-server:latest
...
+
+_Notification:_
+- method: `'yaml/supportSchemaSelection'`
+- params: `void` 
+
+#### SchemaStoreInitialized Notification
+
+The schema store initialized notification is sent from the server to a client to inform client that server finish initializing/loading schemas from schema store, and client now can ask for schemas.
```
Comment (gorkem) [inline README.md:271]: ... that server finish  -> ...that server **has finished**

File: test/schemaSelectionHandlers.test.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+    const result = await selection.getSchemas(testTextDocument.uri);
+
+    expect(result).length(1);
+    expect(result[0]).to.be.eqls({
+      uri: 'https://some.com/some.json',
+      name: 'Schema name',
+      description: 'Schema description',
+    });
+  });
```
Comment (gorkem) [inline test/schemaSelectionHandlers.test.ts:98]: Should it also have a test for the case where there are no schemas associated.

**Review by gorkem — CHANGES_REQUESTED** [review]

File: test/schemaSelectionHandlers.test.ts (hunk 5 above)
Comment (evidolob) [reply test/schemaSelectionHandlers.test.ts:98]: Done

**Review by gorkem — APPROVED** [review]

**Pull request merged**